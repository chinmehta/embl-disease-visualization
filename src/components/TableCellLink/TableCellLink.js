import React from "react";

function TableCellLink({ link, text }) {
  return (
    <React.Fragment>
      <td className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3">
        <a href={link} target="_blank">
          {text}
        </a>
      </td>
    </React.Fragment>
  );
}

export default TableCellLink;
