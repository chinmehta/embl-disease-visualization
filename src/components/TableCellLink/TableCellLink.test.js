import React from "react";
import renderer from "react-test-renderer";

import TableCellLink from "./TableCellLink";

it("renders when no text and link is passed as prop", () => {
  const element = renderer.create(<TableCellLink />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when only text is passed as prop", () => {
  const element = renderer.create(<TableCellLink text="EMBL" />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when only link is passed as prop", () => {
  const element = renderer
    .create(<TableCellLink link="https://www.embl.org" />)
    .toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when both link and text is passed as prop", () => {
  const element = renderer
    .create(<TableCellLink link="https://www.embl.org" text="EMBL" />)
    .toJSON();

  expect(element).toMatchSnapshot();
});
