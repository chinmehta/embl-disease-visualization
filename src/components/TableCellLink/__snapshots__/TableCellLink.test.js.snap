// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when both link and text is passed as prop 1`] = `
<td
  className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
>
  <a
    href="https://www.embl.org"
    target="_blank"
  >
    EMBL
  </a>
</td>
`;

exports[`renders when no text and link is passed as prop 1`] = `
<td
  className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
>
  <a
    target="_blank"
  />
</td>
`;

exports[`renders when only link is passed as prop 1`] = `
<td
  className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
>
  <a
    href="https://www.embl.org"
    target="_blank"
  />
</td>
`;

exports[`renders when only text is passed as prop 1`] = `
<td
  className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
>
  <a
    target="_blank"
  >
    EMBL
  </a>
</td>
`;
